### PR TITLE
build: update forge-std to v1.8.1

### PR DESCRIPTION
**Pull request by smol-ninja**
### Changelog
- Removes `@prb/test` dependency.
- Updates `forge-std` to v1.8.1.
- Since foundry native assertions are declared as `pure`, some of the test functions have also been declared as `pure` .
- Since the visibility of `vm` is private in `StdAssertions`, an instance of `vm` has been added in `Base_Test`.
Ref: https://github.com/foundry-rs/forge-std/issues/530
- Defines `MIN_INT256` and `MAX_INT256` in `Base_Test`.
- Adds `gas-custom-errors: "off"` in `.solhint.json`. 
Ref: https://github.com/protofire/solhint/issues/570

### CODE REVIEW

File: test/Base.t.sol [hunk 1]
```diff
@@ -19,14 +20,25 @@ abstract contract Base_Test is PRBTest, StdCheats, PRBMathAssertions, PRBMathUti
...
                                      CONSTANTS
     //////////////////////////////////////////////////////////////////////////*/
 
+    int256 internal constant MAX_INT256 = type(int256).max;
+
     uint128 internal constant MAX_UINT128 = type(uint128).max;
 
     uint128 internal constant MAX_UINT40 = type(uint40).max;
 
+    int256 internal constant MIN_INT256 = type(int256).min;
```
Comment (PaulRBerg) [inline test/Base.t.sol:40]: Let's order this alphabetically.

**Review by PaulRBerg — APPROVED** [review]
Thanks for this PR and the clear changelog report, @smol-ninja. Also, kudos to your open-source citizenship.

> Defines `MIN_INT256` and `MAX_INT256` in `Base_Test`

Why? Does Forge Std no longer provide them?

File: test/Base.t.sol (hunk 1 above)
Comment (smol-ninja) [reply test/Base.t.sol:40]: > Why? Does Forge Std no longer provide them?

Forge std never provided them. They were [defined in PRBTest](https://github.com/PaulRBerg/prb-test/blob/bdc94959599f18ff135e54615e2eb02b93abba63/src/PRBTest.sol#L60C1-L67C60).

> Let's order this alphabetically.

It's already ordered alphabetically.
- MAX_INT256
- MAX_UINT128
- MAX_UINT40
- MIN_INT256

Comment (PaulRBerg) [reply test/Base.t.sol:40]: > Forge std never provided them. They were defined in PRBTest

Oooh

> It's already ordered alphabetically.

Missed the `MIN` prefix!

**Pull request merged**